src/drivers/cmakeFileApiDriver.ts

```diff
@@ -221,36 +221,46 @@ export class CMakeFileApiDriver extends CMakeDriver {
         return 0;
     }
 
-    async doConfigure(args_: string[], outputConsumer?: proc.OutputConsumer, showCommandOnly?: boolean): Promise<number> {
-        const api_path = this.getCMakeFileApiPath();
+    async doConfigure(args_: string[], outputConsumer?: proc.OutputConsumer, showCommandOnly?: boolean, configurePreset?: ConfigurePreset | null): Promise<number> {
+        const api_path = this.getCMakeFileApiPath(configurePreset?.binaryDir);
         await createQueryFileForApi(api_path);
 
         // Dup args so we can modify them
         const args = Array.from(args_);
-
-        // -S and -B were introduced in CMake 3.13 and this driver assumes CMake >= 3.15
-        args.push(`-S${util.lightNormalizePath(this.sourceDir)}`);
-        args.push(`-B${util.lightNormalizePath(this.binaryDir)}`);
-
-        const gen = this.generator;
         let has_gen = false;
         for (const arg of args) {
             if (arg.startsWith("-DCMAKE_GENERATOR:STRING=")) {
                 has_gen = true;
             }
         }
-        if (!has_gen && gen) {
-            args.push('-G');
-            args.push(gen.name);
-            if (gen.toolset) {
-                args.push('-T');
-                args.push(gen.toolset);
-            }
-            if (gen.platform) {
-                args.push('-A');
-                args.push(gen.platform);
+        const binaryDir = configurePreset?.binaryDir ? configurePreset.binaryDir : this.binaryDir;
```

Comment:
It should be simplified to `configurePreset?.binaryDir ?? this.binaryDir`

(if `??` doesn't work, use `||` instead)